lib/components/normal-components/PinHeader.ts

```diff
@@ -19,16 +19,31 @@ export class PinHeader extends NormalComponent<typeof pinHeaderProps> {
     const holeDiameter = this._parsedProps.holeDiameter
     const platedDiameter = this._parsedProps.platedDiameter
     const pitch = this._parsedProps.pitch
+    const showSilkscreenPinLabels = this._parsedProps.showSilkscreenPinLabels
+
     if (pinCount > 0) {
+      let footprintString: string
+
       if (pitch) {
         if (!holeDiameter && !platedDiameter) {
-          return `pinrow${pinCount}_p${pitch}`
+          footprintString = `pinrow${pinCount}_p${pitch}`
+        } else {
+          footprintString = `pinrow${pinCount}_p${pitch}_id${holeDiameter}_od${platedDiameter}`
+        }
+      } else {
+        if (!holeDiameter && !platedDiameter) {
+          footprintString = `pinrow${pinCount}`
+        } else {
+          return null
         }
-        return `pinrow${pinCount}_p${pitch}_id${holeDiameter}_od${platedDiameter}`
       }
-      if (!holeDiameter && !platedDiameter) {
-        return `pinrow${pinCount}`
+
+      // Add _hidesilkscreenpinlabels if showSilkscreenPinLabels is false or undefined
```

Comment:
why does the comment not match the implementation? `nopinlabels` vs `_hidesilkscreenpinlabels`